pool.go

```diff
@@ -126,10 +127,14 @@ func newPool(c *Config) *pool {
 		refresh:   make(chan struct{}, 1),
 		done:      make(chan struct{}, 1),
 	}
-	go p.refreshPool()
+
 	return &p
 }
 
+func (p *pool) Start() {
+	go p.refreshPool()
+}
+
```

Comment:
what do we get by factoring this out as a separate API to call versus having it implicit in `newPool`?

Reply:
@willscott The flexibility to start the refresh loop whenever we want in tests instead of always having to do it on creation of the pool.